### DISCUSSION TITLE
How to update record attacher with existing image?

### DISCUSSION THREAD

**Discussion by kobius** [shrinerb/shrine#695] [Q&A] [answered]
All of my images are hosted on S3 and accessible only via Cloudfront signed URLs. This works perfectly in Shrine.

I have a model `Book` that can have a `book_cover_image`. When creating a new Book, this field is `nil`, and I have a background worker that checks via an API and receives a filename if a cover image already exists, something like: `covers/85/book-title-1.png`.

These images are all already stored on my S3 bucket that is setup to work with my app via Shrine.

How can I update an existing `Book` object filename to use an existing image on my bucket, without duplicating the image?

**Comment by janko** [comment, marked as answer]
See https://github.com/shrinerb/shrine/discussions/688. It's a bit hidden in the documentation, I should somehow surface it better in the docs.

**Reply by kobius** [reply]
Perfect, thanks @janko 🙏

**Reply by jrochkind** [reply]
it's hard to doc everything prominently when there is so much to doc! (cause shrine is so rich).  

It is mentioned in the guide on Attacher at https://shrinerb.com/docs/attacher#changes , but maybe it could get it's own heading or be more prominent -- but I don't know if people would find it in this Guide anyhow, hard to say!
